packages/extension/src/companion/CompanionPermission.tsx

```diff
@@ -4,18 +4,18 @@ import {
 } from '@dailydotdev/shared/src/components/buttons/Button';
 import classed from '@dailydotdev/shared/src/lib/classed';
 import { companionExplainerVideo } from '@dailydotdev/shared/src/lib/constants';
-import React, { ReactElement, Ref, forwardRef } from 'react';
+import React, { ReactElement, Ref, forwardRef, useContext } from 'react';
 import PlayIcon from '@dailydotdev/shared/src/components/icons/Play';
 import { ClickableText } from '@dailydotdev/shared/src/components/buttons/ClickableText';
-import { useExtensionPermission } from '@dailydotdev/shared/src/hooks';
+import { ExtensionContext } from '@dailydotdev/shared/src/contexts/common';
 
 const CompanionSection = classed('div', 'flex flex-col max-w-full');
 
 const CompanionPermissionComponent = (
   _,
   ref: Ref<HTMLDivElement>,
 ): ReactElement => {
-  const { requestContentScripts } = useExtensionPermission();
+  const { requestContentScripts } = useContext(ExtensionContext);
```

Comment:
Perhaps export as direct use hook, it's a common practice we recently started doing.
Example being: `useAnalyticsContext`